apps/common/behaviours/casework-submission.js

```diff
@@ -1,15 +1,16 @@
 'use strict';
 
+const AuthToken = require('../models/auth-token');
 const CaseworkModel = require('../models/i-casework');
 const StatsD = require('hot-shots');
 const client = new StatsD();
 
 const Compose = func => superclass => class extends superclass {
   prepare() {
-    if (typeof func === 'function') {
-      return Object.assign(super.prepare(), func(this.toJSON()));
-    }
-    return super.prepare();
+    const model = new AuthToken();
+    return model.auth().then(token => {
```

Comment:
Prepare statement is now async, so we have to handle other prepare statements calls elsewhere in the service as if they are async too just in case. This is also how this step is treated in the HOF model anyway. Here we generate a token bearer and pass it to the payload builder functions and the prepare step essentially in the `apps/common/models/i-casework.js` model where we can update all urls at once.